contrib/clojure-package/examples/cnn-text-classification/src/cnn_text_classification/classifier.clj

```diff
@@ -0,0 +1,112 @@
+            :sentence-size sentence-size
+            :embedding-size embedding-size})
+  (let [shuffled (shuffle (map (fn [d l] [d l]) data label))
+        train-num (- (count shuffled) test-num)
+        training (into [] (take train-num shuffled))
+        test (into [] (drop train-num shuffled))]
+    {:training {:data  (ndarray/array (into [] (flatten (mapv (fn [v] (first v)) training)))
+                                      [train-num 1 sentence-size embedding-size]) ;; has to be channel x y
+                :label (ndarray/array (into [] (flatten (mapv (fn [v] (last v) ) training)))
+                                      [train-num])}
+     :test {:data  (ndarray/array (into [] (flatten (mapv (fn [v] (first v)) test)))
+                                  [test-num 1 sentence-size embedding-size]) ;; has to be channel x y
+            :label (ndarray/array (into [] (flatten (mapv (fn [v] (last v) ) test)))
+                                  [test-num])}}))
+
+;;; convnet with multiple filter sizes
+;; from Convolutional Neural Networks for Sentence Classification by Yoon Kim
+(defn get-multi-filter-convnet [num-embed sentence-size batch-size]
+  (let [filter-list [3 4 5]
+        num-filter 100
+        num-label 2
+        dropout 0.5
+        input-x (sym/variable "data")
+        polled-outputs (mapv (fn [filter-size]
+                               (as-> (sym/convolution {:data input-x
+                                                       :kernel [filter-size num-embed]
+                                                       :num-filter num-filter}) data
+                                 (sym/activation {:data data :act-type "relu"})
+                                 (sym/pooling {:data data
+                                               :pool-type "max"
+                                               :kernel [(inc (- sentence-size filter-size)) 1]
+                                               :stride [1 1]})))
+                             filter-list)
+        total-filters (* num-filter (count filter-list))
+        concat (sym/concat "concat" nil polled-outputs {:dim 1})
+        hpool (sym/reshape "hpool" {:data concat :target-shape [batch-size total-filters]})
+        hdrop (if (> dropout 0) (sym/dropout "hdrop" {:data hpool :p dropout}) hpool)
+        fc (sym/fully-connected  "fc1" {:data hdrop :num-hidden num-label})]
+    (sym/softmax-output "softmax" {:data fc})))
+
+(defn train-convnet [{:keys [devs embedding-size batch-size test-size num-epoch max-examples]}]
+  (let [glove (data-helper/load-glove glove-file-path) ;; you can also use word2vec
+        ms-dataset (data-helper/load-ms-with-embeddings mr-dataset-path embedding-size glove max-examples)
+        sentence-size (:sentence-size ms-dataset)
+        shuffled (shuffle-data test-size ms-dataset)
+        train-data (mx-io/ndarray-iter [(get-in shuffled [:training :data])]
+                                       {:label[(get-in  shuffled [:training :label])]
```

Comment:
we need a space between key and the value.
And there is an extra space between get-in and params